fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCQueryBuilder.java

```diff
@@ -1255,44 +1255,44 @@ private SqlQueryData processTokenParm(Class<?> resourceType, QueryParameter quer
+                                    .replace(UNDERSCORE_WILDCARD, ESCAPE_UNDERSCORE) + PERCENT_WILDCARD;
+                            bindVariables.add(SearchUtil.normalizeForSearch(textSearchString));
+                            appendEscape = true;
                         } else {
-                            // If no code system specified, search against both normalized code and unmodified code.
-                            whereClauseSegment.append(tableAlias + DOT).append(TOKEN_VALUE);
-                            if (NE.equals(operator)) {
-                                whereClauseSegment.append(NOT);
+                            // Determine code system case-sensitivity
+                            boolean codeSystemIsCaseSensitive = false;
+                            if (value.getValueSystem() != null && !value.getValueSystem().isEmpty()) {
+                                whereClauseSegment.append(tableAlias + DOT).append(TOKEN_VALUE).append(operator).append(BIND_VAR);
+
+                                // Normalize code if code system is not case-sensitive. Otherwise leave code as is. 
+                                codeSystemIsCaseSensitive = CodeSystemSupport.isCaseSensitive(value.getValueSystem());
+                                bindVariables.add(SqlParameterEncoder.encode(codeSystemIsCaseSensitive ?
+                                        value.getValueCode() : SearchUtil.normalizeForSearch(value.getValueCode())));
+                            } else {
+                                // If no code system specified, search against both normalized code and unmodified code.
+                                whereClauseSegment.append(tableAlias + DOT).append(TOKEN_VALUE);
+                                if (NE.equals(operator)) {
+                                    whereClauseSegment.append(NOT);
+                                }
+                                whereClauseSegment.append(IN).append(LEFT_PAREN).append(BIND_VAR).append(COMMA).append(BIND_VAR).append(RIGHT_PAREN);
+                                bindVariables.add(SqlParameterEncoder.encode(value.getValueCode()));
+                                bindVariables.add(SqlParameterEncoder.encode(SearchUtil.normalizeForSearch(value.getValueCode())));
                             }
-                            whereClauseSegment.append(IN).append(LEFT_PAREN).append(BIND_VAR).append(COMMA).append(BIND_VAR).append(RIGHT_PAREN);
-                            bindVariables.add(SqlParameterEncoder.encode(value.getValueCode()));
-                            bindVariables.add(SqlParameterEncoder.encode(SearchUtil.normalizeForSearch(value.getValueCode())));
                         }
                     }
 
                     // Include system if present.
                     if (value.getValueSystem() != null && !value.getValueSystem().isEmpty()) {
                         Long commonTokenValueId = null;
-                        if (NE.equals(operator)) {
```

Comment:
It is no longer possible to have an NE operator on a token parm. Previously it was used in the case of the `:not` modifier, but since the switch to use a WHERE NOT EXISTS in that case, token parms don't use NE